python/cudf/cudf/core/column/methods.py

```diff
@@ -0,0 +1,65 @@
+import cudf
+
+
+class ColumnMethodsMixin:
+    def _return_or_inplace(self, new_col, **kwargs):
+        """
+        Returns an object of the type of the column owner or updates the column
+        of the owner (Series or Index) to mimic an inplace operation
```

Comment:
Hmm -- good catch. There shouldn't be a way for `DataFrame` or `MultiIndex` objects to reach here, but this isn't the best place to validate that. I'll think about this a bit more.

Reply:
Since we only define `.str`, `.cat`, etc., for `Index` and `Series`, it's not possible for `self._parent` to ever be a `DataFrame` or `MultiIndex`. I think we should be OK here without checking for anything